modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/xmpp/XMPPClient.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl.util.xmpp;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jivesoftware.smack.ConnectionConfiguration;
+import org.jivesoftware.smack.PacketListener;
+import org.jivesoftware.smack.SmackConfiguration;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.filter.AndFilter;
+import org.jivesoftware.smack.filter.FromContainsFilter;
+import org.jivesoftware.smack.filter.PacketFilter;
+import org.jivesoftware.smack.filter.PacketTypeFilter;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.packet.Packet;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+
+/**
+ * This class contains the Agent specific implementation for all the XMPP functionality. This
+ * includes connecting to a XMPP Server & Login using the device's XMPP-Account, Setting
+ * listeners and filters on incoming XMPP messages and Sending XMPP replies for control signals
+ * received. Makes use of the 'Smack-XMPP' library provided by jivesoftware/igniterealtime.
+ * <p/>
+ * It is an abstract class with an abstract method 'processXMPPMessage' allowing the user to have
+ * their own implementation of the actions to be taken upon receiving an appropriate XMPP message.
+ */
+public abstract class XMPPClient {
+	private static final Log log = LogFactory.getLog(XMPPClient.class);
+
+	private int replyTimeoutInterval = 500;    // millis
```

Comment:
better to make this a CLASS VARIABLE or even CONSTANT if value is not going to be changed.